docs/dev/README.md

```diff
@@ -0,0 +1,70 @@
+# Quick start guide for Assisted Service Developer
+
+Over time, assisted-service has been adapted to different scenarios, and has become one more piece of many other subsystems in which it works integrated with other Openshift components.
+
+For this reason, depending on the development we are working on, we have different scenarios for developers that allow us greater flexibility for our daily work. For example, there will be cases in which we are only interested in doing some specific tests on the API and this can be covered with a light environment, but in other cases we may need a complete Openshift cluster with its control plane and worker nodes, in which case the scenario will be more complex and heavy.
+
+The frequently used scenarios are described below.
+
+## Testing simple changes to the REST service
+[Detailed documentation](../../deploy/podman/README.md)
+
+In this scenario the goal is to run small tests with the assisted-installer REST API. This can be done in a local environment by starting the assisted-service with podman.
+
+We can start it directly with make:
+```shell
+make deploy-onprem
+```
+
+## Testing e2e-flow with REST
+[Detailed documentation](https://github.com/openshift/assisted-test-infra/blob/master/README.md)
+
+This scenario is useful for developing and testing our end-to-end tests.
+This environment requires a large amount of resources since it will deploy additional baremetal hosts (VMs), so **running it in local environments is not recommended**.
+
+We can start this scenario from the `assisted-test-infra` project:
+```shell
+make run
```

Comment:
I have been working on a guide for this use case, it's a workflow that I use regularly.

Feel free to use as much or as little as you like to add to this.

I am a big fan of using `aicli` and `kcli`, two great utilities writtern by our own @karmab so I have included setup information for my favoured use case in this doc.

Note that `aicli` and `kcli` are not official RedHat tools ... But they are superb :)

https://spaces.redhat.com/display/AI/Installing+a+cluster+with+assisted-test-infra%2C+kcli+and+aicli+and+HTTP+API

Reply:
We really should mention that `assisted-test-infra` requires setup before you can perform `make run`

Also we should make it clear that assisted-test-infra is very opinionated during the setup phase and that the ideal place to use it is on a Beaker box or other ephemeral server. Using this on your laptop could cause some pain.

Reply:
> I am a big fan of using aicli and kcli, two great utilities writtern by our own @karmab so I have included setup information for my favoured use case in this doc.

I agree with Paul, should we mention those tools in the docs? We mention "tests" a lot, and I belive they are implied to be manual most of the time, `aicli` is a nice tool to do that, and `kcli` also is worth a mention IMO